draft-ietf-quic-transport.md

```diff
@@ -1922,13 +1922,14 @@ validation is performed both during connection establishment (see
 
 Connection establishment implicitly provides address validation for both
 endpoints.  In particular, receipt of a packet protected with Handshake keys
-confirms that the client received the Initial packet from the server.  Once the
-server has successfully processed a Handshake packet from the client, it can
-consider the client address to have been validated.
-
-Additionally, a server MAY consider the client address validated if the
-client uses a connection ID chosen by the server and the connection ID contains
-at least 64 bits of entropy.
+confirms that the peer successfully processed an Initial packet.  Once an
+endpoint has successfully processed a Handshake packet from the peer, it can
+consider the peer address to have been validated.
+
+Additionally, an endpoint MAY consider the peer address validated if the peer
+uses a connection ID chosen by the endpoint and the connection ID contains at
+least 64 bits of entropy.  This means that any valid packet received by the
```

Comment:
I think there is a disconnect between the two sentences.

First sentence talks about the peer using the connection ID chosen by the endpoint. But the second sentence tries to derive from the first sentence that the original DCID chosen by the client, which is _never_ chosen as a connection ID by the server, working as a proof.

I think it might make sense to preserve the second paragraph as-is, then add a separate paragraph pointing out that a client validates the server's address when receiving a Initial packet that can be successfully processed, because the protection keys are derived from the client-chosen DCID that is at least 64 bits long.

Reply:
This is a good point.  I've a proposal inbound that doesn't add sentences, but rewrites the second instead.

Retry or Version Negotiation are included, but by virtue of their inclusion as a Source Connection ID, not the Initial packet protection.